src/models/db/utils.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
+
+const assert = require('assert')
+const knex = require('../../lib/knex').knex
+const InvalidModificationError = require('five-bells-shared')
+  .InvalidModificationError
+
+function createModule (tableName, convertToPersistent, convertFromPersistent) {
+  function getTransaction (options) {
+    return options && options.transaction ? options.transaction : knex
+  }
+
+  function select (where, transaction) {
+    return (transaction || knex).from(tableName).select().where(where)
+      .then((results) => results.map(convertFromPersistent))
+  }
+
+  function selectOne (where, transaction) {
+    return (transaction || knex).from(tableName).select().where(where)
+      .then((results) => {
+        if (results.length === 1) {
+          return convertFromPersistent(results[0])
+        } else if (results.length === 0) {
+          return null
+        } else {
+          assert(false, 'Multiple rows in ' + tableName + ' match ' +
+            JSON.stringify(where))
+        }
+      })
+  }
+
+  function update (data, where, transaction) {
+    return (transaction || knex)(tableName)
+      .update(convertToPersistent(data)).where(where)
+  }
+
+  function insert (data, transaction) {
+    return (transaction || knex).insert(convertToPersistent(data))
+      .into(tableName)
+  }
+
+  function insertAll (data, transaction) {
+    return (transaction || knex).insert(data.map(convertToPersistent))
+      .into(tableName)
+  }
+
+  function _upsert (data, where, transaction) {
+    assert(transaction, 'transaction is required for upsert')
+    return selectOne(where, transaction).then((existing) => {
```

Comment:
The `upsert` method uses `existed`. Might make sense to also name this `existed` for consistency.